vip_hci/leastsq/leastsq.py

```diff
@@ -163,15 +162,15 @@ def _leastsq_ann(array, indices, ann, n_annuli, fwhm, angles, delta_rot, metric,
     pa_threshold, _, _ = _define_annuli(angles, ann, n_annuli, fwhm, radius_int,
                                         asize, delta_rot, n_segments_ann,
                                         verbose)
-    res = []
     if nproc == 1:
+        res = []
         for j in range(n_segments_ann):
             segm_res = _leastsq_patch(array, j, indices, angles, pa_threshold,
                                       metric, dist_threshold, solver, tol)
             res.append(segm_res)
 
     elif nproc > 1:
-        pool = Pool(processes=int(nproc))
+        pool = Pool(processes=nproc)
```

Comment:
processes argument of Pool takes an intenger. It's safer to cast nproc as an int here

Reply:
As long as `cpu_count()` returns an integer, `nproc` is also an integer because of the floor division `//` in new line 113. I propose to remove the redundant casting.

Reply:
I see to options: you either leave it with the integer casting or you use the floor division in each function that uses multiprocessing in VIP.

Reply:
I favour the latter... I will create a new pull request once this one is done, as it is not strictly related to the `xloci`.